src/Functions/FunctionDateOrDateTimeToSomething.h

```diff
@@ -83,6 +83,13 @@ class FunctionDateOrDateTimeToSomething : public IFunctionDateOrDateTime<Transfo
                 arguments[0].type->getName(), this->getName());
     }
 
+    bool hasInformationAboutPreimage() const override { return Transform::hasPreimage(); }
+
+    RangeOrNull getPreimage(const IDataType & type, const Field & point) const override
+    {
+        return Transform::getPreimage(type, point);
```

Comment:
we can define this function like
```
if constexpr (Transform::hasPregimage()) {return getPreimage ...}
else return std::nullopt;
```
so that we don't have to declear so many `getPreimage`

Of course, use `static constexpr bool hasPreimage` for each Transform class

Reply:
Exactly! This constexpr-if statement would save a lot of code size. Thanks for your suggestion!

My only concern is the return of `std::nullopt` would lead to the silent errors when `FunctionDateOrDateTimeToSomething::hasInformationAboutPreimage() == false` and  `FunctionDateOrDateTimeToSomething::getPreimage()` is called. Maybe we should throw an exception here as we did for other similar cases?